Pre-existing test failure: test_monitor_loop_gives_up_after_max_retries times out

## Bug: Pre-existing Test Failure on main

**Test:** `tests/regression/test_issue_40_error_swallowing.py::test_monitor_loop_gives_up_after_max_retries`

This test fails on `main` — it is **not** caused by sm#187 (dispatch templates).

## Spec

See `docs/working/209_monitor_loop_test_timeout.md` — read before implementing.

## Failure Mode

The test times out at the pytest-timeout limit (30s). The monitor loop performs 5 retries with exponential backoff, but the sleep/backoff is not mocked, causing real-time waits that exceed the timeout.

```
FAILED tests/regression/test_issue_40_error_swallowing.py::test_monitor_loop_gives_up_after_max_retries
E   Failed: Timeout (>30.0s) from pytest-timeout.
```

## Root Cause

`_monitor_loop()` (`src/message_queue.py:756`) calls `asyncio.sleep(retry_delay)` between retries with no mock in the test. With `initial_retry_delay=1.0` and exponential doubling, 5 retries consume 1+2+4+8+16=31s real time, exceeding the 30s pytest-timeout.

## Fix

Pass tiny `initial_retry_delay_seconds` via config when constructing `MessageQueueManager` in the test, so retries complete in ~5ms. Replace the 35s sleep with a short polling loop. See spec for full diff sketch.

## Reproduction

```bash
source venv/bin/activate
python -m pytest tests/regression/test_issue_40_error_swallowing.py::test_monitor_loop_gives_up_after_max_retries -v
```

--- a/tests/regression/test_issue_40_error_swallowing.py
+++ b/tests/regression/test_issue_40_error_swallowing.py
@@ -399,7 +399,7 @@ async def test_monitor_loop_gives_up_after_max_retries(tmp_path, caplog):
     queue_mgr = MessageQueueManager(
         session_manager=mock_session_manager,
         db_path=str(tmp_path / "test.db"),
-        config={"input_poll_interval": 0.05}
+        config={"timeouts": {"message_queue": {"initial_retry_delay_seconds": 0.001}}},
     )
 
     # Start monitoring first
@@ -411,10 +411,12 @@ async def test_monitor_loop_gives_up_after_max_retries(tmp_path, caplog):
 
     queue_mgr._get_sessions_with_pending = always_fail
 
-    # Wait for retries to exhaust
-    # Backoffs: 1s, 2s, 4s, 8s, 16s = ~31s total
+    # real retries complete in ~5ms; poll until loop gives up
     with caplog.at_level("ERROR"):
-        await asyncio.sleep(35)
+        for _ in range(100):
+            if "giving up" in caplog.text:
+                break
+            await asyncio.sleep(0.05)
 
     # Stop monitoring
     await queue_mgr.stop()

